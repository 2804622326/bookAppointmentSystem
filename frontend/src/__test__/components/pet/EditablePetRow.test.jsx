import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EditablePetRow from '../../../components/pet/EditablePetRow.jsx';

test('edits and saves pet', () => {
  const onSave = jest.fn();
  const onCancel = jest.fn();
  const pet = { id: 1, name: 'a', type: 'dog', breed: 'b', color: 'c', age: 2 };
  const { asFragment } = render(
    <table><tbody><EditablePetRow pet={pet} index={0} onSave={onSave} onCancel={onCancel} /></tbody></table>
  );
  fireEvent.change(screen.getByDisplayValue('a'), { target: { value: 'x' } });
  fireEvent.click(screen.getAllByRole('button')[0]);
  expect(onSave).toHaveBeenCalled();
  fireEvent.click(screen.getAllByRole('button')[1]);
  expect(onCancel).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
