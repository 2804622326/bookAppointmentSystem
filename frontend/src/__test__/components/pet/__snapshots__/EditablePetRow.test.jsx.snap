// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edits and saves pet 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          <input
            class="form-control"
            name="name"
            type="text"
            value="x"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="type"
            type="text"
            value="dog"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="breed"
            type="text"
            value="b"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="color"
            type="text"
            value="c"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="age"
            type="number"
            value="2"
          />
        </td>
        <td>
           
          <button
            class="me-2 btn btn-success btn-sm"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.97 4.97a.75.75 0 0 1 1.07 1.05l-3.99 4.99a.75.75 0 0 1-1.08.02L4.324 8.384a.75.75 0 1 1 1.06-1.06l2.094 2.093 3.473-4.425z"
              />
            </svg>
          </button>
        </td>
        <td
          colspan="2"
        >
          <button
            class="btn btn-secondary btn-sm"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
